src/main/java/mekanism/common/item/gear/ItemAtomicDisassembler.java

```diff
@@ -184,45 +176,15 @@ public boolean onBlockStartBreak(ItemStack stack, BlockPos pos, Player player) {
                 //Even though we now handle breaking bounding blocks properly, don't allow vein mining them
                 // only allow mining things that are considered an ore
                 if (!(state.getBlock() instanceof BlockBounding) && state.is(MekanismTags.Blocks.ATOMIC_DISASSEMBLER_ORE)) {
-                    List<BlockPos> found = findPositions(state, pos, world);
-                    MekanismUtils.veinMineArea(energyContainer, world, pos, (ServerPlayer) player, stack, this, found,
-                          hardness -> getDestroyEnergy(baseDestroyEnergy, hardness), distance -> 0.5 * Math.pow(distance, 1.5), state);
+                    Map<BlockPos, BlockState> found = ModuleVeinMiningUnit.findPositions(world, Map.of(pos.immutable(), state), false, -1);
```

Comment:
Is this pos actually mutable at times? And if so does it even matter given we aren't persisting it nor modifying it ourselves?